RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -546,16 +544,18 @@ void MahiFit::phase1Debug(const HBHEChannelInfo& channelData, MahiDebugInfo& mdi
         mdi.itPulse[iTS] = nnlsWork_.pulseMat.col(iBX).coeff(iTS);
       }
     } else if (nnlsWork_.bxs.coeff(iBX) == pedestalBX_) {
+      mdi.pedEnergy = 0;
       mdi.pedEnergy = nnlsWork_.ampVec.coeff(iBX);
-    } else if (nnlsWork_.bxs.coeff(iBX) == -1) {
-      mdi.pEnergy = nnlsWork_.ampVec.coeff(iBX);
+    } else if (nnlsWork_.bxs.coeff(iBX) < 8) {
+      int ootIndex = nnlsWork_.bxs.coeff(iBX);
+      if (ootIndex > 0)
+        ootIndex += 2;
+      else
+        ootIndex += 3;
+      mdi.ootEnergy[ootIndex] = 0;
```

Comment:
shouldn't there be a check or assert here that ootIndex is in the range from 0 to 6 (it can also be checked on L549)

Also, is the math above correct? If I got it right we have here 0 -> 3 and 1 -> 3. It may be OK if 0 is not possible.

Reply:
Now restricted the range to be -3 to 4 in L549. 
```
    } else if (nnlsWork_.bxs.coeff(iBX)>=-3 && nnlsWork_.bxs.coeff(iBX) <= 4) {
```

Math is correct because 0 is not possible: 
``` 
    if (nnlsWork_.bxs.coeff(iBX) == 0) {
   ...
    } else if (nnlsWork_.bxs.coeff(iBX) == pedestalBX_) {
   ...
    } else if (nnlsWork_.bxs.coeff(iBX)>=-3 && nnlsWork_.bxs.coeff(iBX) <= 4) {
```